staging/src/k8s.io/apiextensions-apiserver/test/integration/validation_test.go

```diff
@@ -801,21 +801,24 @@ spec:
 	}
 
 	// wait for condition to go away
-	t.Log("Wait for condition to disappear")
+	t.Log("Wait for schema validation condition to disappear")
 	err = wait.PollImmediate(100*time.Millisecond, 5*time.Second, func() (bool, error) {
 		obj, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
 		cond = findCRDCondition(obj, apiextensionsv1beta1.NonStructuralSchema)
-		return cond == nil, nil
+		if v := "spec.versions[0].schema.openAPIV3Schema.properties[a].type: Required value: must not be empty for specified object fields"; strings.Contains(cond.Message, v) {
+			return false, nil
+		}
+		return true, nil
```

Comment:
would add another step where you set preserveUnknownFields to false. Then this last violation should also go.

Reply:
If I set preserveUnknownFields to false, I get a validation error that says the all fields in schema will be pruned which I think is a valid error. Instead, I tried fixing both the validation errors with a patch. Now there will be no violations